components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/mediators/XMLSchemaValidator.java

```diff
@@ -0,0 +1,277 @@
+        if (messageProperty != null) {
+            entityExpansionLimit = Integer.parseInt(messageProperty.toString());
+
+        } else {
+            String message = "XML schema entityExpansionLimit property value is missing.";
+            ThreatExceptionHandler.handleException(messageContext, message);
+        }
+
+        messageProperty = messageContext.getProperty(ThreatProtectorConstants.CHILDREN_PER_ELEMENT);
+        if (messageProperty == null) {
+            String message = "XML schema childrenElement property value is missing.";
+            ThreatExceptionHandler.handleException(messageContext, message);
+        } else {
+            childrenPerElement = Integer.parseInt(messageProperty.toString());
+        }
+
+        if (log.isDebugEnabled()) {
+            log.debug(("DTD enable:" + dtdEnabled) + ", " + "External entities: " + externalEntitiesEnabled
+                    + ", " + "Element Count:" + elementCount + ", " + "Max AttributeLength:" + attributeLength
+                    + ", " + "Max xml Depth:" + maxXMLDepth + ", " + "Attribute count:" + attributeCount + ", "
+                    + "Entity Expansion Limit" + attributeCount + ". " + "childrenElement:" + attributeCount);
+        }
+        XMLConfig xmlConfig = new XMLConfig();
+        xmlConfig.setDtdEnabled(dtdEnabled);
+        xmlConfig.setExternalEntitiesEnabled(externalEntitiesEnabled);
+        xmlConfig.setMaxDepth(maxXMLDepth);
+        xmlConfig.setMaxElementCount(elementCount);
+        xmlConfig.setMaxAttributeCount(attributeCount);
+        xmlConfig.setMaxAttributeLength(attributeLength);
+        xmlConfig.setEntityExpansionLimit(entityExpansionLimit);
+        xmlConfig.setMaxChildrenPerElement(childrenPerElement);
+        return xmlConfig;
+    }
+
+    /**
+     * If the isContentAware method returns false, The request message payload wont be build.
+     *
+     * @return isContentAware method always returns false to avoid build the message.
+     */
+    @Override
+    public boolean isContentAware() {
+        return false;
+    }
+
+    /**
+     * This method validates the request payload xml with the relevant xsd.
+     *
+     * @param messageContext This message context contains the request message properties of the relevant
+     *                       API which was enabled the XML_Validator message mediation in flow.
+     * @return This method returns the success or not (true/false) status of the schema validation.
+     */
+    private boolean validateSchema(MessageContext messageContext, BufferedInputStream bufferedInputStream)
+            throws APIMThreatAnalyzerException {
+        String xsdURL;
+        Schema schema;
+        SchemaFactory schemaFactory = SchemaFactory.newInstance(XMLConstants.W3C_XML_SCHEMA_NS_URI);
+        try {
+            Object messageProperty = messageContext.getProperty(APIMgtGatewayConstants.XSD_URL);
+            if (messageProperty == null) {
+                return true;
```

Comment:
Since we are checking isEmpty in following block, can't we use stringutils.isempty , which will include both null and empty check

Reply:
I can use the StringUtils.isEmpty. but even I use it I have to do the null check since it is an object.

Reply:
it checks for both null and empty internally.

Reply:
StringUtils.isEmpty normally checks
whether the string is null or not
but in here I want to check the object.